tree/treeplayer/inc/TMPWorkerTree.h

```diff
@@ -35,8 +35,9 @@ class TMPWorkerTree : public TMPWorker {
    /// \endcond
 public:
    TMPWorkerTree();
-   TMPWorkerTree(const std::vector<std::string>& fileNames, const std::string& treeName, unsigned nWorkers, ULong64_t maxEntries);
-   TMPWorkerTree(TTree *tree, unsigned nWorkers, ULong64_t maxEntries);
+   TMPWorkerTree(const std::vector<std::string> &fileNames, TEntryList *entries, const std::string &treeName,
```

Comment:
Yes. But then we have to add something in TEntryList to test is usability. All codes use the fact that 'entries == nullptr' means no entry list, and 'entries != nullptr' means a valid entry list. We need something like 'entries.IsValid()' or '!entries.IsNull()' .